File: C++/soln-array-problems/2sum.cpp
/**
 @author Farheen Bano

 Date 05-08-2021
  
 Reference-
 https://leetcode.com/problems/two-sum/
*/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        if(nums.size()<2)
            return {};
        unordered_map<int,int> map;
        for(int i=0;i<nums.size();i++){
            int other=target-nums[i];
            if(map.find(nums[i])!= map.end())
                return {map[nums[i]],i};
            map[other]=i;
        }
        return {};
    }
};

File: Python/soln-array-problems/2sum.py
"""
 @author Farheen Bano

 Date 11-08-2021
  
 Reference-
 https://leetcode.com/problems/two-sum/
"""

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums)<2:
            return []
        record=dict()
        for i in range(len(nums)):
            other=target-nums[i]
            if nums[i] in record:
                return [record[nums[i]],i]
            record[other]=i
        return []
        